presto-main/src/main/java/com/facebook/presto/server/security/SpnegoFilter.java

```diff
@@ -165,7 +165,8 @@ public Principal getUserPrincipal()
                         return;
                     }
                 }
-                catch (GSSException e) {
+                catch (Exception e) {
+                    LOG.debug(e, "Authentication failed for token %s", parts[1]);
```

Comment:
Log-and-throw is an anti-pattern. Either log or throw, but never both. Otherwise, you end up with the same exception getting logged multiple places is slightly different ways which is very confusing.

If the original exception is getting lost somewhere, we should figure that out.